src/csharp/unitypackage/unitypackage_skeleton/Plugins/Grpc.Core/runtimes/grpc_csharp_ext_dummy_stubs.c

```diff
@@ -0,0 +1,116 @@
+
+// Copyright 2019 The gRPC Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// When building for Unity Android with il2cpp backend, Unity tries to link
+// the __Internal PInvoke definitions (which are required by iOS) even though
+// the .so/.dll will be actually used. This file provides dummy stubs to
+// make il2cpp happy.
+// See https://github.com/grpc/grpc/issues/16012
+
+void grpcsharp_init() {}
```

Comment:
It doable to log and abort in all of these stub functions?

Reply:
In general it is, but I didn't want do add a dependency on grpc libraries in order to log and abort in a portable way (this might end up in the static dummy library linking grpc). I'll give it a thought and if I come up with some reasonable way to abort, I'll do it.

Reply:
Added an fprintf message and abort().